packages/cli/src/workflows/workflows.services.ts

```diff
@@ -0,0 +1,33 @@
+import { FindOneOptions } from 'typeorm';
+import { Db } from '..';
+import { SharedWorkflow } from '../databases/entities/SharedWorkflow';
+import { User } from '../databases/entities/User';
+
+export class WorkflowsService {
+	static async getSharing(
+		user: User,
+		workflowId: number | string,
+		relations: string[] | undefined = ['workflow'],
+		{ allowGlobalOwner } = { allowGlobalOwner: true },
+	): Promise<SharedWorkflow | undefined> {
+		const options: FindOneOptions = {
+			where: {
+				workflow: { id: workflowId },
+			},
+		};
+
+		// Omit user from where if the requesting user is the global
+		// owner. This allows the global owner to view and delete
+		// workflows they don't own.
+		if (!allowGlobalOwner || user.globalRole.name !== 'owner') {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```

Comment:
You can use a generic to avoid the exception:
```ts
const options: FindOneOptions<SharedWorkflow> & { where: ObjectLiteral } = {
	where: {
		workflow: { id: workflowId },
	},
};
```